### ISSUE TITLE
Houdini: Reset JOB environment

### ISSUE THREAD

**Issue by mustafa-zarkash** [ynput/OpenPype#5517]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

`$JOB` always has the same value of `$HIP` 
Unless you save the hip file in another asset!, it keeps the old `$JOB` 
This can be solved by this enhancement:

![image](https://github.com/ynput/OpenPype/assets/20871534/84002699-7709-4cf8-9095-dcbb6bfaf151)

### How would you imagine the implementation of the feature?

by implementing function `set_job_path()`

```python
import os
import hou
job = os.environ["HIP"]
hou.hscript("set JOB=" + job)
```

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

**Comment by BigRoy** [comment]
I wouldn't make this a manual menu entry in the menu to be honest since it's easily forgotten by an artist and still means that just opening a scene might not be entirely right, yes? - this should be either automatic or studio-managed?

E.g. we have `JOB `set to a specific resources folder of the project, not particularly to the current asset.

**Comment by krishna8008** [comment]
Thanks for bringing this discussion @mustafa-zarkash 

I agree with @BigRoy  here, this should be automaticlly changed when we saving file in different task.
Also I would like to ask if possible we can save all hip files in "hip"folder, so the files will be more organised. IMO the better way of handling files is

1. sh001 should be the path for JOB
2. inside we can create hip file and save tasks there.
This way, even if the other department people want to access the files in the same shot, cross department people can use $JOB environment and access all task project files from there.

![image](https://github.com/ynput/OpenPype/assets/61413765/0ebc4d12-dc5e-4830-b910-81110dee2d57)

**Comment by BigRoy** [comment]
> Also I would like to ask if possible we can save all hip files in "hip"folder, so the files will be more organised. IMO the better way of handling files is

I feel like this is more of a workfile template issue then related to this issue. It's a good question, but probably be rephrased and asked on the community forum since I believe there are ways to currently do so.

> sh001 should be the path for JOB

That's roughly similar to what we do - but not exactly the same.

**Comment by krishna8008** [comment]
> I agree with @BigRoy here, this should be automaticlly changed when we saving file in different task. 



If we can automate the $JOB variable to autoset that should solve the problem.

**Comment by mustafa-zarkash** [comment]
> this should be either automatic or studio-managed?

I agree, this makes more sense! 

>  For us if `JOB`==`HIP` then why use `JOB`?

I have the same question actually! 
but let me rephrase my thought : 
1. What `JOB` value should be ? 
2. even if `JOB` is redundant , it shouldn't have the value of another asset path!

**Comment by BigRoy** [comment]
> What JOB value should be ?

As explained, it's likely studios would have their own requirements for that?

> even if JOB is redundant , it shouldn't have the value of another asset path!

I agree with that - that it'd be quite confusing to operate like that yes. 🗡️ So we might want to at least have a mechanism in-place to allow updating it on task change.

I recall @fabiaserra having done something with that in the past and actually sharing some code snippets about that.

**Comment by mustafa-zarkash** [comment]
> As explained, it's likely studios would have their own requirements for that?

Maybe we can expose it to settings and leave it to studios to decide! IDK, maybe we can post it in another issue.

**Comment by fabiaserra** [comment]
isn't $JOB already automatically defined in Houdini to be the parent folder of $HIP?

**Comment by mustafa-zarkash** [comment]
> isn't $JOB already automatically defined in Houdini to be the parent folder of $HIP?

Yes, However, when you open asset `A` and save file in asset `B` 
`$JOB` of asset `B` still points to the asset `A` path! 

I think there should be a post save action that sets `$JOB` if doesn't match `$HIP` for example.

**Comment by fabiaserra** [comment]
then we should just need to add this in a `456.py` file
```
hou.allowEnvironmentToOverwriteVariable("JOB", True)
```
?
https://www.sidefx.com/docs/houdini/hom/hou/allowEnvironmentToOverwriteVariable.html

**Comment by fabiaserra** [comment]
> then we should just need to add this in a `456.py` file
> 
> ```
> hou.allowEnvironmentToOverwriteVariable("JOB", True)
> ```
> 
> ? https://www.sidefx.com/docs/houdini/hom/hou/allowEnvironmentToOverwriteVariable.html

nvm, I see that would be only if $JOB actually changed on the env... maybe we just explicitly set it on the 456.py to whatever context it loads? it would probably just match $AVALON_WORKDIR?

**Comment by mustafa-zarkash** [comment]
> maybe we just explicitly set it on the 456.py to whatever context it loads?

I think it will be easier to add that inside 
https://github.com/ynput/OpenPype/blob/fd64e261df0371b16395aa5238fee1eb6f0fd705/openpype/hosts/houdini/api/pipeline.py#L307-L314
